xarray/core/dataarray.py

```diff
@@ -130,18 +130,30 @@
     T_XarrayOther = TypeVar("T_XarrayOther", bound="DataArray" | Dataset)
 
 
-def _check_coords_dims(shape, coords, dim):
-    sizes = dict(zip(dim, shape, strict=True))
+def _check_coords_dims(
+    shape: tuple[int, ...], coords: Coordinates, dims: tuple[Hashable, ...]
```

Comment:
Only internal so low stress, but FYI we agreed to standardize on `dim` for param names (except for constructor)